storage/src/nodestore/persist.rs

```diff
@@ -281,6 +330,18 @@ impl NodeStore<Arc<ImmutableProposal>, FileBacked> {
 
         let flush_start = Instant::now();
 
+        let mut header = self.header;
+        let mut node_allocator = NodeAllocator::new(self.storage.as_ref(), &mut header);
+
+        // Collect all unpersisted nodes first to avoid mutating self while iterating
+        let unpersisted_nodes: Vec<MaybePersistedNode> = {
+            let unpersisted_iter = UnPersistedNodeIterator::new(self);
+            unpersisted_iter.collect()
+        };
```

Comment:
bit weird to do this in two lines.
```suggestion
        // Collect all unpersisted nodes first to avoid mutating self while iterating
        let unpersisted_nodes = UnPersistedNodeIterator::new(self).collect::<Vec<_>>();
```